### ISSUE TITLE
H264 Encoded frames are jumbled(not ordered properly) in the dump file

### ISSUE THREAD

**Issue by APrashant** [Intel-Media-SDK/MediaSDK#2847]
## System information
**CPU information**
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              12
On-line CPU(s) list: 0-11
Thread(s) per core:  2
Core(s) per socket:  6
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               158
Model name:          Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
Stepping:            10
CPU MHz:             800.511
CPU max MHz:         4600.0000
CPU min MHz:         800.0000
BogoMIPS:            6399.96
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            12288K
NUMA node0 CPU(s):   0-11

**GPU information**
00:00.0 Host bridge [0600]: Intel Corporation 8th Gen Core Processor Host Bridge/DRAM Registers [8086:3ec2] (rev 07)
00:02.0 VGA compatible controller [0300]: Intel Corporation Device [8086:3e92]
00:08.0 System peripheral [0880]: Intel Corporation Xeon E3-1200 v5/v6 / E3-1500 v5 / 6th/7th Gen Core Processor Gaussian Mixture Model [8086:1911]
00:12.0 Signal processing controller [1180]: Intel Corporation Cannon Lake PCH Thermal Controller [8086:a379] (rev 10)
00:14.0 USB controller [0c03]: Intel Corporation Cannon Lake PCH USB 3.1 xHCI Host Controller [8086:a36d] (rev 10)
00:14.2 RAM memory [0500]: Intel Corporation Cannon Lake PCH Shared SRAM [8086:a36f] (rev 10)
00:15.0 Serial bus controller [0c80]: Intel Corporation Device [8086:a368] (rev 10)
00:16.0 Communication controller [0780]: Intel Corporation Cannon Lake PCH HECI Controller [8086:a360] (rev 10)
00:17.0 RAID bus controller [0104]: Intel Corporation SATA Controller [RAID mode] [8086:2822] (rev 10)
00:1c.0 PCI bridge [0604]: Intel Corporation Device [8086:a33b] (rev f0)
00:1d.0 PCI bridge [0604]: Intel Corporation Cannon Lake PCH PCI Express Root Port 9 [8086:a330] (rev f0)
00:1f.0 ISA bridge [0601]: Intel Corporation Device [8086:a306] (rev 10)
00:1f.3 Audio device [0403]: Intel Corporation Cannon Lake PCH cAVS [8086:a348] (rev 10)
00:1f.4 SMBus [0c05]: Intel Corporation Cannon Lake PCH SMBus Controller [8086:a323] (rev 10)
00:1f.5 Serial bus controller [0c80]: Intel Corporation Cannon Lake PCH SPI Controller [8086:a324] (rev 10)
00:1f.6 Ethernet controller [0200]: Intel Corporation Ethernet Connection (7) I219-LM [8086:15bb] (rev 10)
01:00.0 PCI bridge [0604]: Texas Instruments XIO2001 PCI Express-to-PCI Bridge [104c:8240]

msdk version: latest cloned master with "[vpp] Removed SelectThreadDependencyPattern call" last commit (21.4.3)
libva info: VA-API version 1.13.0
Multi Transcoding Sample Version 8.4.27.0

**Issue behavior**
current behavior:
We tried to dump the h264 encoding file with the following mentioned par file. 
We have analysed the dumped h264 files and observed that the frames are not reordered properly. 
ie., we took a 1min timer video as input (which decrease from 60 - 0 ) and dumped to a h264 file. where in dumped h264 file, some seconds are shuffled instead of playing in decreased order. However test passed.

**sample_multi_transcode Par file used:**
-i::h264 1min.h264 -o::sink
-hw -i::source -o::h264 wo_comp1.h264
-hw -i::source -o::h264 wo_comp2.h264

**Input file(1min.h264) and dump files**
I have added input file used and dump files got in this zip. Kindly look into it.
[input_and_dump.zip](https://github.com/Intel-Media-SDK/MediaSDK/files/7711473/input_and_dump.zip)

**same behavior for below par file too:**
-i::h264 1min.h264 -join -hw -async 1 -vpp_comp_dst_x 0 -vpp_comp_dst_y 0 -vpp_comp_dst_w 1280 -vpp_comp_dst_h 720 -o::sink -ext_allocator
-vpp_comp 1 -async 1 -join -hw -i::source -o::h264 wi_comp_1min1.h264 

Please help us in resolving this issue. 

Note: we also tried with sample_encode - same thing observed (frame reordering) in h264 dump.
Description used: ./sample_encode h264 -i vpp_dump_file_thread.yuv -o encode_output.h264 -w 1280 -h 720

Thanks in advance.

**Comment by APrashant** [comment]
Hi,

With only h264 dump we are facing this frame reordering and frame dropping issue.
Could you please give any inputs for debugging this issue.

Thanks in advance.

**Comment by dmitryermilov** [comment]
Please say near which frames you observe wrong ordering?  I looked at both wo_comp1.h264/wo_comp2.h264 and don't see reordering issues in comparison to 1min.h264 source.

**Comment by APrashant** [comment]
Hi @dmitryermilov 

we have checked with the VLC media player using following command.
**vlc --demux h264 wo_comp1_latest.h264**

We have observed the time is not decremented sequentially. Instead we got following timings. 
00:29:00, 25, 24, 23, 22, 21, 27, 26, 25, 14
00:25:00
00:24:00
00:23:00
00:22:00
00:21:00
**Instead of getting 00:20:00 got following timings**
00:27:00
00:26:00
00:25:00
00:14:00

Thanks.
